src/modules/graph/graph.service.test.ts

```diff
@@ -24,84 +25,92 @@ describe('GraphService', () => {
-      mockSuccessfulGraphApiCall();
-      mockSuccessfulGraphGetCall();
-      when(mockRequest.filter).calledWith(filterStr).mockReturnValueOnce(mockRequest);
-
-      await graphService.get<string>({ path, filter: filterStr });
-
-      expect(mockRequest.expand).toHaveBeenCalledTimes(0);
-    });
-
-    it('chains input parameters as api methods and returns the response', async () => {
-      mockSuccessfulGraphApiCall();
-      mockSuccessfulGraphGetCall();
-      when(mockRequest.filter).calledWith(filterStr).mockReturnValueOnce(mockRequest);
-      when(mockRequest.expand).calledWith(expandStr).mockReturnValueOnce(mockRequest);
+    it('calls all graph client methods on a graph serviec get request with multiple parameters and returns the response', async () => {
+      mockSuccessfulCompleteGraphRequest();
 
       const result = await graphService.get<string>({ path, filter: filterStr, expand: expandStr });
 
-      expect(mockClient.api).toHaveBeenCalledTimes(1);
-      expect(mockRequest.filter).toHaveBeenCalledTimes(1);
-      expect(mockRequest.expand).toHaveBeenCalledTimes(1);
-      expect(mockRequest.get).toHaveBeenCalledTimes(1);
-
-      expect(mockClient.api).toHaveBeenCalledWith(path);
-      expect(mockRequest.filter).toHaveBeenCalledWith(filterStr);
-      expect(mockRequest.expand).toHaveBeenCalledWith(expandStr);
-      expect(mockRequest.get).toHaveBeenCalledWith();
+      expectGraphMethodsToHaveBeenCalled({
+        apiCalled: true,
+        filterCalled: true,
+        expandCalled: true,
+        getCalled: true,
+      });
 
       expect(result).toEqual(expectedResponse);
     });
   });
 
-  const mockSuccessfulGraphApiCall = () => when(mockClient.api).calledWith(path).mockReturnValueOnce(mockRequest);
-  const mockSuccessfulGraphGetCall = () => when(mockRequest.get).calledWith().mockResolvedValue(expectedResponse);
+  const mockSuccessfulGraphApiCall = () => mockGraphClientService.mockSuccessfulGraphApiCallWithPath(path);
+
+  const mockSuccessfulChainingCalls = () => {
+    mockGraphClientService.mockSuccessfulExpandCallWithExpandString(expandStr).mockSuccessfulFilterCallWithFilterString(filterStr);
+  };
+
+  const mockSuccessfulGraphGetCall = () => mockGraphClientService.mockSuccessfulGraphGetCall(expectedResponse);
+
+  const mockSuccessfulCompleteGraphRequest = () => {
+    mockSuccessfulGraphApiCall();
+    mockSuccessfulChainingCalls();
+    mockSuccessfulGraphGetCall();
+  };
+
+  const expectGraphMethodsToHaveBeenCalled = ({
+    apiCalled = false,
+    filterCalled = false,
+    expandCalled = false,
+    getCalled = false,
+  }: {
+    apiCalled?: boolean;
+    filterCalled?: boolean;
+    expandCalled?: boolean;
+    getCalled?: boolean;
+  }) => {
+    apiCalled
+      ? (expect(mockGraphClientService.client.api).toHaveBeenCalledTimes(1), expect(mockGraphClientService.client.api).toHaveBeenCalledWith(path))
+      : expect(mockGraphClientService.client.api).toHaveBeenCalledTimes(0);
```

Comment:
Personally I don't like the "evaluate an expression that evaluates two void functions in a tuple and then throw away the tuple" trick that we're doing here to make this syntax work, I think it's too surprising to be executing void functions in this way. I'd prefer us to just use if statements

Or we could refactor this function to look like
```
const expectations = getCallExpectations({ apiCalled, filterCalled, expandCalled, getCalled });
expectations.forEach((expectation) => expectation());
```
where `getCallExpectations` looks something like
```
const apiCallExpectations = apiCalled
  ? [ () => expect(mockGraphClientService.client.api).toHaveBeenCalledTimes(1), () => expect(mockGraphClientService.client.api).toHaveBeenCalledWith(path)]
  : [ () => expect(mockGraphClientService.client.api).toHaveBeenCalledTimes(0)]
// etc.
```